nixos/modules/services/torrent/btpd.nix

```diff
@@ -0,0 +1,171 @@
+        default = null;
+        description = ''
+          IP address to send to trackers.
+          Null sends no address, so trackers will advertise the one they see btpd connect from.
+        '';
+      };
+      port = mkOption {
+        type = types.port;
+        default = 6681;
+        description = ''
+          Port for btpd to listen on.
+        '';
+      };
+      ipv4 = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to enable IPv4 support for btpd.
+        '';
+      };
+      ipv6 = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable IPv6 support for btpd.
+        '';
+      };
+      bandwidthLimitIn = mkOption {
+        type = types.ints.unsigned;
+        default = 0;
+        description = ''
+          Limit incoming BitTorrent traffic to n kB/s. 0 is unlimited.
+        '';
+      };
+      bandwidthLimitOut = mkOption {
+        type = types.ints.unsigned;
+        default = 0;
+        description = ''
+          Limit outgoing BitTorrent traffic to n kB/s. 0 is unlimited.
+        '';
+      };
+      emptyStart = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Start btpd without any active torrents.
+        '';
+      };
+      ipcPermission = mkOption {
+        type = types.ints.unsigned;
+        default = 0600;
+        description = ''
+          Permission mode of the btpd socket.
+        '';
+      };
+      maxPeers = mkOption {
+        type = types.ints.unsigned;
+        default = 0;
+        description = ''
+          Limit the amount of peers to n.
+        '';
+      };
+      maxUploads = mkOption {
+        type = types.int;
+        default = -2;
+        description = ''
+          Controls the number of simultaneous uploads.
+          The possible values are:
+          <literallayout>
+          n &lt; -1 : Choose n >= 2 based on bandwidthLimitOut (default).
+          n = -1 : Upload to every interested peer.
+          n =  0 : Don't upload to anyone.
+          n &gt;  0 : Upload to at most n peers simultaneously.
+          </literallayout>
+        '';
+      };
+      prealloc = mkOption {
+        type = types.ints.unsigned;
+        default = 2048;
+        description = ''
+          Preallocate disk space in chunks of n kB.
+          Note that n will be rounded up to the closest multiple of the
+          torrent piece size. If n is zero no preallocation will be done.
+        '';
+      };
+    };
+  };
+  config = mkIf cfg.enable {
+    systemd.services.btpd = {
+      after = [ "network.target" ];
+      description = "BitTorrent Protocol Daemon";
+      wantedBy = [ "multi-user.target" ];
+      path = [ cfg.package ];
+      serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        Type = "forking";
+        PIDFile = "${cfg.baseDir}/pid";
```

Comment:
Usually it is preferred to run programs in the foreground instead of having them be forked. Unless there's a good reason for this, please change this. This will also have the benefit that logs will be viewable via system (the standard way on NixOS) and not some log file.